tests/providers/azure/azure_test.py

```diff
@@ -238,6 +257,77 @@ def testCreateDiskFromSnapshot(self, mock_create_disk):
+                                    mock_get_container,
+                                    mock_get_blob_client,
+                                    mock_blob_storage,
+                                    mock_create_container):
+    """Test that a disk can be created from a snapshot URI."""
+    mock_create_disk.return_value.done.return_value = True
+    mock_create_disk.return_value.result.return_value = MOCK_DISK_COPY
+    mock_create_storage_account.return_value = ('fake-account-id', 'fake-key')
+    mock_blob_client.return_value = None
+    mock_blob_storage.return_value = None
+    mock_get_container.return_value = mock.Mock()
+    mock_create_container.return_value = None
+    blob_properties = mock_get_blob_client.return_value.get_blob_properties
+    blob_properties.return_value = mock.Mock(copy=mock.Mock(status='success'))
+    mock_delete_storage_account.return_value = None
+
+    disk_from_snapshot_uri = FAKE_ACCOUNT.CreateDiskFromSnapshotURI(
+        FAKE_SNAPSHOT, 'fake-snapshot-uri')
+    #  hashlib.sha1('/a/b/c/fake-resource-group/fake_snapshot_name'.encode(
+    #     'utf-8')).hexdigest()[:23] = bff00b08549ba8b975b2e70
+    mock_create_storage_account.assert_called_with(
+        'bff00b08549ba8b975b2e70', region='fake-region')
+    self.assertIsInstance(disk_from_snapshot_uri, compute.AZDisk)
+    self.assertEqual(
+        'fake_snapshot_name_f4c186ac_copy', disk_from_snapshot_uri.name)
+    mock_create_disk.assert_called_with(
+        FAKE_SNAPSHOT.resource_group_name,
+        'fake_snapshot_name_f4c186ac_copy',
+        mock.ANY,
+        sku='Standard_LRS')
+
+  @mock.patch('azure.mgmt.resource.subscriptions.v2019_11_01.operations._subscriptions_operations.SubscriptionsOperations.list')
+  @typing.no_type_check
+  def testListSubscriptionIDs(self, mock_list):
+    """Test that subscription IDs are correctly listed"""
+    mock_list.return_value = MOCK_LIST_IDS
+    subscription_ids = FAKE_ACCOUNT.ListSubscriptionIDs()
+    self.assertEqual(2, len(subscription_ids))
+    self.assertEqual('fake-subscription-id-1', subscription_ids[0])
+
+  @mock.patch('azure.mgmt.storage.v2019_06_01.operations._storage_accounts_operations.StorageAccountsOperations.list_keys')
+  @mock.patch('azure.mgmt.storage.v2019_06_01.operations._storage_accounts_operations.StorageAccountsOperations.create')
+  @typing.no_type_check
+  def testCreateStorageAccount(self, mock_create, mock_list_keys):
+    """Test that a storage account is created and its information retrieved"""
+    # pylint: disable=protected-access
+    mock_create.return_value.result.return_value = MOCK_STORAGE_ACCOUNT
+    mock_list_keys.return_value = MOCK_LIST_KEYS
+    account_id, account_key = FAKE_ACCOUNT._CreateStorageAccount('fakename')
+    self.assertEqual('fakestorageid', account_id)
+    self.assertEqual('fake-key-value', account_key)
+
+    with self.assertRaises(ValueError):
+      _, _ = FAKE_ACCOUNT._CreateStorageAccount(
+          'fake-non-conform-name')
```

Comment:
You can also check that the error string matches what you expect (since the error is pretty broad)

```suggestion
    with self.assertRaises(ValueError) as error:
      _, _ = FAKE_ACCOUNT._CreateStorageAccount(
          'fake-non-conform-name')
    # str(error.exception) will contain the exception string. You can access it *outside* the with statement
```

Reply:
Added in all statements, makes it clearer to the reader